internal/commands/exec/exec_test.go

```diff
@@ -3,27 +3,44 @@ package exec
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"fmt"
+	"net"
 	"net/http"
 	"net/http/httptest"
 	"net/url"
 	"os"
 	"path"
+	"strconv"
 	"strings"
 	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
 
 	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/proto-public/pbdataplane"
 	"github.com/hashicorp/go-hclog"
 
+	"github.com/hashicorp/consul-api-gateway/internal/consul"
 	gwTesting "github.com/hashicorp/consul-api-gateway/internal/testing"
 )
 
-func TestRunExecLoginError(t *testing.T) {
-	t.Parallel()
+type testDPServer struct{}
```

Comment:
Do we not actually need to mock the [`WatchServers`](https://github.com/hashicorp/consul/blob/main/proto-public/pbserverdiscovery/serverdiscovery.proto#L13) method anywhere? Are we only exercising that in e2e tests?

Reply:
No, we don't -- this is just a gRPC server that stubs out the endpoints. So the library actually hits this and issues real API calls against this. It's almost like using `httptest`.